### PR TITLE
issue #2 - added redis database handler

### PR DESCRIPTION
**Pull request by macskay**
Pastes get commited to redis database with random generated hashkey.

Resolving issue #2

### CODE REVIEW

**Comment by macskay** [comment]
grafik

**Comment by macskay** [comment]
Beware: No db configuration added yet, meaning pastes dont get deleted after a fixed time period and no password for the server is set.

File: pasteit/routes/index.js [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
+var client = redis.createClient();
+var flatten = require("flat")
 
-router.use(bodyParser.urlencoded({ extended : false }));
-router.use(bodyParser.json());
-router.use(function(err, req, res, next) {
-    console.error(err.stack);
-    res.status(500).send("Something broke!");
-});
+var getNewHash = function() {
```
Comment (ghost) [inline pasteit/routes/index.js:9]: Why not use a UUID4?

File: pasteit/routes/index.js [hunk 2]
```diff
@@ -1,24 +1,45 @@
...
+
+    client.hmset(hash, new_paste);
+
+    res.redirect("/"+hash);  
+});
+
+router.get("/:hash/", function(req, res) {
+    var hash = req.params.hash;
+
+    var result = client.hgetall(hash, function(err, reply) {
```
Comment (ghost) [inline pasteit/routes/index.js:37]: What will hgetall return if hash is not found?

**Review by ghost — APPROVED** [review]
Overall this looks good. I tested the functionality locally on a fresh install of redis. You could merge this, but think about my comments.

File: pasteit/routes/index.js (hunk 2 above)
Comment (macskay) [reply pasteit/routes/index.js:37]: oh right, we would need a 404 or sth like that. I'll look into that.

File: pasteit/routes/index.js (hunk 1 above)
Comment (macskay) [reply pasteit/routes/index.js:9]: We can use that as well, no need to use my function. I'll patch it up.

Comment (macskay) [reply pasteit/routes/index.js:9]: UUID4 is now used

grafik

File: pasteit/routes/index.js (hunk 2 above)
Comment (macskay) [reply pasteit/routes/index.js:37]: Upon entering a wrong hash an error page is shown.

grafik

**Pull request merged**